src/main/java/com/dougnoel/sentinel/steps/APISteps.java

```diff
@@ -150,37 +150,65 @@ public static void verifyResponseTime(double time) {
-	 * 
+	 * <p>
+	 * <b>Gherkin Examples:</b>
+	 * <ul>
+	 * <li>I validate the response <b>contains</b> the text <b>"key":"value"</b></li>
+	 * <li>I validate the response <b>does not contain</b> the text <b>"value":"key"</b></li>
+	 * <li>I validate the response <b>has</b> the text <b>123456789ABCDEFG</b></li>
+	 * </ul>
+	 * <p>
 	 * @param assertion String null to see if the text exists, "does not" to see if it is absent
 	 * @param matchType String use "contains" for a partial match otherwise it will be an exact match
 	 * @param text String the text to match
 	 */
-	@Then("^I validate the response( does not)? (has|have|contains?) the text \"([^\"]*)\"$")
-    public static void verifyResponseContains(String assertion, String matchType, String text) {
-        boolean negate = !StringUtils.isEmpty(assertion);
-        boolean partialMatch = matchType.contains("contain");
+	@Then("^I validate the response( does not)? (has|have|contains?) the text (.*?)$")
+	public static void verifyResponseContains(String assertion, String matchType, String text) {
+		boolean negate = !StringUtils.isEmpty(assertion);
+		boolean partialMatch = matchType.contains("contain");
 
-        int responseCode = APIManager.getResponse().getResponseCode();
-        String responseText = APIManager.getResponse().getResponse();
-        String expectedResult = SentinelStringUtils.format(
-                "Expected the response to {}{} the text {}. The response had a response code of {} and contained the text: {}",
-                (negate ? "not " : ""), (partialMatch ? "contain" : "exactly match"), text, responseCode, responseText
-                        .replace("\n", " "));
-        log.trace(expectedResult);
-        if (partialMatch) {
-            if (negate) {
-                assertFalse(expectedResult, responseText.contains(text));
-            } else {
-                assertTrue(expectedResult, responseText.contains(text));
-            }
-        } else {
-            if (negate) {
-                assertFalse(expectedResult, StringUtils.equals(responseText, text));
-            } else {
-                assertTrue(expectedResult, StringUtils.equals(responseText, text));
-            }
-        }
-    }
+		int responseCode = APIManager.getResponse().getResponseCode();
+		String responseText = APIManager.getResponse().getResponse();
+		String expectedResult = SentinelStringUtils.format(
+			"Expected the response to {}{} the text {}. The response had a response code of {} and contained the text: {}",
+			(negate ? "not " : ""), (partialMatch ? "contain" : "exactly match"), text, responseCode, responseText
+				.replace("\n", " "));
+		log.trace(expectedResult);
+
+		boolean negateResult;
+		boolean result;
+		boolean textStartEndQuotes = false;
+		String startEndQuoteStripText = text;
+		if(text.matches("^\".*?\"$")) {
+			textStartEndQuotes = true;
+			startEndQuoteStripText = text.substring(1, text.length() - 1);
+		}
+
+		if (partialMatch) {
```

Comment:
To reduce code complexity you'll need to break this out into a different function, and at that point it should also be applied to TextVerificationSteps.java to have them follow this new pattern.
Sorry to increase your scope!